code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/GithubOAuthImpl.java

```diff
@@ -41,7 +42,7 @@ public Account getAccount(ApiRequest request) {
             return null;
         }
         String token = request.getServletContext().getRequest().getHeader(AUTH_HEADER);
-        if (StringUtils.isEmpty(token)) {
+        if (StringUtils.isEmpty(token) || !token.startsWith(AUTH_TYPE)) {
```

Comment:
@alena1108 This class should return an account if **either** there is a JWT in the `token` param OR in a  `Authorization: Bearer jwtdata` header.  So return immediately is no good.

The problem @hibooboo2 is fixing is that in certain cases the browser running the UI generates requests that contains both a `Authorization: Basic adsfasdf` and `?token=jwtdata` query param.  We want to take the query param and ignore the `Authorization` header in this class, because it is not a Bearer token, but it was not checking that part before so it never looked at the `token` query param.